contracts/staking-v1/src/tests.rs

```diff
@@ -2203,4 +2203,174 @@ fn test_lockups_migration_v100() {
     assert_eq!(staker_info.lockups, vec![]);
     assert_eq!(staker_info.unlocked_stake_amount, Uint128::from(200u128));
     assert_eq!(staker_info.locked_stake_amount, Uint128::zero());
+
+    // test query stakers with legacy lockups and update method
+    let mut deps = mock_dependencies(&[]);
+    let msg = InstantiateMsg {
+        owner: "owner".to_string(),
+        bro_token: "bro0000".to_string(),
+        rewards_pool_contract: "reward0000".to_string(),
+        bbro_minter_contract: "bbrominter0000".to_string(),
+        epoch_manager_contract: "epoch0000".to_string(),
+        community_bonding_contract: Some("community_bonding0000".to_string()),
+        unstake_period_blocks: 10,
+        min_staking_amount: Uint128::zero(),
+        min_lockup_period_epochs: 1,
+        max_lockup_period_epochs: 365,
+        base_rate: Decimal::from_str("0.0001").unwrap(),
+        linear_growth: Decimal::from_str("0.0005").unwrap(),
+        exponential_growth: Decimal::from_str("0.0000075").unwrap(),
+    };
+
+    let info = mock_info("addr0000", &[]);
+    let _res = instantiate(deps.as_mut(), env.clone(), info, msg).unwrap();
+
+    let stakers = vec![
+        deps.as_mut().api.addr_canonicalize("addr0000").unwrap(),
+        deps.as_mut().api.addr_canonicalize("addr0001").unwrap(),
+        deps.as_mut().api.addr_canonicalize("addr0002").unwrap(),
+        deps.as_mut().api.addr_canonicalize("addr0003").unwrap(),
+    ];
+
+    crate::state::store_staker_info(
+        deps.as_mut().storage,
+        &stakers[0],
+        &StakerInfo {
+            reward_index: Decimal::zero(),
+            unlocked_stake_amount: Uint128::zero(),
+            locked_stake_amount: Uint128::from(100u128),
+            pending_bro_reward: Uint128::zero(),
+            pending_bbro_reward: Uint128::zero(),
+            last_balance_update: 0,
+            lockups: vec![LockupInfo {
+                amount: Uint128::from(100u128),
+                unlocked_at: Expiration::AtHeight(1200),
+                locked_at_block: None,
+                epochs_locked: None,
+            }],
+        },
+    )
+    .unwrap();
+    crate::state::store_staker_info(
+        deps.as_mut().storage,
```

Comment:
does a function make sense instead of repeating 4 times?

Reply:
it is a test, implementing a function in the contract just to write 4 stakers might be not the best idea